### PR TITLE
✨ Generate automatic language names for docs translations

### PR DESCRIPTION
**Pull request by jakul**
This PR does two things:

1) Update the code that generates the mkdocs.yml files to always include a local language name

    The local language name is read from a new file which contains mappings for all the allocated ISO 639-1 language codes, and the content of that file is sourced from [https://quickref.me/iso-639-1](https://quickref.me/iso-639-1), though with updates to persist the existing language names where there was a difference with the existing name for the language in the FastAPI source.
    
    ⚠️ The old code always put English at the top of the language list in the UI, but the new code doesn't
    
    | Before | After |
    | ------------- | ------------- |
    | Captura de Pantalla 2022-09-06 a las 13 05 30 | Captura de Pantalla 2022-09-06 a las 13 03 10 | 
   

2) Regenerate all the mkdocs.yaml files

### CODE REVIEW

File: scripts/docs.py [hunk 1]
```diff
@@ -369,25 +370,34 @@ def update_config(lang: str):
...
         config["theme"]["language"] = current_config["theme"]["language"]
-    languages = [{"en": "/"}]
-    alternate: List[Dict[str, str]] = config["extra"].get("alternate", [])
-    alternate_dict = {alt["link"]: alt["name"] for alt in alternate}
-    new_alternate: List[Dict[str, str]] = []
+
+    languages: List[Dict[str, str]] = []
+    alternate: List[Dict[str, str]] = []
+
+    # Language names sourced from https://quickref.me/iso-639-1. FastAPI
```
Comment (MrRawbin) [inline scripts/docs.py:377]: This is a very minor complaint, but the dot in the end of the sentence breaks the URL.

**Review by MrRawbin — APPROVED** [review]
Very cool! Good job.

File: scripts/docs.py (hunk 1 above)
Comment (jakul) [reply scripts/docs.py]: Ok, I've added a quick commit to fix it.

**Comment by jakul** [comment]
> Very cool! Good job.

Thanks for the quick response!

**Review by BilalAlpaslan — APPROVED** [review]

**Comment by jakul** [comment]
Thanks for the review @BilalAlpaslan.

I noticed that the link to go back to the English version was wrong, it was `/en/` instead of `/`. I've fixed that in a new commit.

**Review by BilalAlpaslan — APPROVED** [review]

**Review by Xewus — APPROVED** [review]

**Comment by Xewus** [comment]
Honestly, if not for you, I would not have known about switching the language at the top of the page. 🤭

**Comment by bharathkalyans** [comment]
Actually, I was about to raise a PR regarding this issue, but found your PR that hasn't been yet merged 🥲.
Also, we have a new emoji language and I think shifting it below would give a better experience. Your take on this?
Screenshot 2023-08-27 at 9 58 53 PM

**Review by creyD — APPROVED** [review]
Thank you for this PR, looks good.

**Review by alejsdev — APPROVED** [review]

**Comment by tiangolo** [comment]
📝 Docs preview for commit 5b1d076c5946f06e7a0cc7eb8b3fd7a9a3abbbbc at: https://7e140f4b.fastapitiangolo.pages.dev

**Comment by tiangolo** [comment]
Great idea, thank you @jakul! This looks great. 🚀 

And thanks for the reviews @creyD, @BilalAlpaslan, @MrRawbin, @Xewus, @alejsdev 🍰 

---

I updated `docs.py` with the recent changes and language setup refactor to use this data, and I updated the file to be YAML instead of JSON, just because all the other config files are YAML too, so, just for consistency. 😅

**Pull request merged**